src/ops/ops.go

```diff
@@ -1033,6 +1033,52 @@ func stripDev(device string) string {
 	return strings.Replace(device, "/dev/", "", 1)
 }
 
+// getPartitionPathFromLsblk uses lsblk to find the actual partition device path
+// for a given device and partition number. This handles all device types including
+// device mapper devices correctly.
+func (o *ops) getPartitionPathFromLsblk(device, partitionNumber string) (string, error) {
+	type node struct {
+		Name     string  `json:"name"`
+		Size     int64   `json:"size"`
+		Children []*node `json:"children"`
+	}
+	var disks struct {
+		Blockdevices []*node `json:"blockdevices"`
+	}
+
+	ret, err := o.ExecPrivilegeCommand(nil, "lsblk", "-b", "-J")
```

Comment:
Avoid short flags in code

```suggestion
	ret, err := o.ExecPrivilegeCommand(nil, "lsblk", "--bytes, "--json")
```

Reply:
Also we can pass lsblk the device and it will give us only the information about that device

```suggestion
	ret, err := o.ExecPrivilegeCommand(nil, "lsblk", "-b", "-J", device)
```